hoprd/rest-api/src/session.rs

```diff
@@ -43,6 +51,159 @@ lazy_static::lazy_static! {
     ).unwrap();
 }
 
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize, utoipa::IntoParams, utoipa::ToSchema)]
+#[into_params(parameter_in = Query)]
+#[serde(rename_all = "camelCase")]
+pub(crate) struct SessionWebsocketClientQueryRequest {
+    #[schema(required = true)]
+    #[serde_as(as = "DisplayFromStr")]
+    pub destination: PeerId,
+    #[schema(required = true)]
+    pub hops: u8,
```

Comment:
```suggestion
    pub path: RoutingOptions
```

See that `RoutingOptions` is fully serde compliant.

Reply:
It is serde compliant, but cannot be deserialized using `serde_home_form` or `serde_urlencode`. I had to remove some capability to retain the overall url parsability. :/